src/ui/Search/searchinfilesworker.cpp

```diff
@@ -22,53 +22,43 @@ SearchInFilesWorker::~SearchInFilesWorker()
 
 void SearchInFilesWorker::run()
 {
-    // Search string converted to a regex
-    QString rawSearch = frmSearchReplace::rawSearchString(m_string, m_searchMode, m_searchOptions);
+    FileSearchResult::SearchResult searchResult;
+    searchResult.search = m_string;
 
     QFlags<QRegularExpression::PatternOption> options = QRegularExpression::NoPatternOption;
-    if (m_searchOptions.MatchCase == false) {
-        options |= QRegularExpression::CaseInsensitiveOption;
-    }
+    QFlags<QDirIterator::IteratorFlag> dirIteratorOptions = QDirIterator::NoIteratorFlags;
 
-    QRegularExpression regex(rawSearch, options);
+    if (m_searchMode == SearchHelpers::SearchMode::Regex) {
+        if (m_searchOptions.MatchCase == false) {
+            options |= QRegularExpression::CaseInsensitiveOption;
+        }
+        QString rawSearch = frmSearchReplace::rawSearchString(m_string, m_searchMode, m_searchOptions);
+        m_regex.setPattern(rawSearch);
+        m_regex.setPatternOptions(options);
+    }else if (m_searchMode == SearchHelpers::SearchMode::SpecialChars) {
+        QString rawSearch = frmSearchReplace::rawSearchString(m_string, m_searchMode, m_searchOptions);
+        m_string = unescapeString(m_string);
```

Comment:
Just pushed a new commit.... care to try that \x4corem test again? 
Pretty sure this should fix it.